File: python/Figuras/figura2d.py
"""
Clase abstracta para figuras bidimensionales
"""
# pylint: disable=invalid-name
from abc import abstractmethod
from Figura import Figura

class Figura2d(Figura):
    """Clase abstracta para figuras bidimensionales"""

    def __init__(self, nombre: str, id_figura: int):
        """
        Constructor de la clase Figura2d

        Args:
            nombre (str): Nombre de la figura
            id_figura (int): Identificador único de la figura
        """
        super().__init__(nombre, "2D", id_figura)

    @abstractmethod
    def calcular_area(self) -> float:
        """Calcula el área de la figura 2D"""

    @abstractmethod
    def calcular_perimetro(self) -> float:
        """Calcula el perímetro de la figura 2D"""

File: java/Figuras/src/main/java/org/example/Figura2d.java
package org.example;

/**
 * Clase abstracta para figuras bidimensionales
 */
public abstract class Figura2d extends Figura {
    
    /**
     * Constructor de la clase Figura2d
     * @param nombre Nombre de la figura
     * @param idFigura Identificador único de la figura
     */
    public Figura2d(String nombre, int idFigura) {
        super(nombre, "2D", idFigura);
    }
    
    /**
     * Calcula el área de la figura 2D
     * @return El área de la figura
     */
    public abstract double calcularArea();
    
    /**
     * Calcula el perímetro de la figura 2D
     * @return El perímetro de la figura
     */
    public abstract double calcularPerimetro();
}